index/get_manifest_info.py

```diff
@@ -76,26 +80,40 @@ def _get_library(manifest_info):
     return library
 
 
+def _get_library_collection_code(manifest_info):
+    """ Get Primo lsr01 Collection Code given repository """
+    library_collection_code = ""
+    if manifest_info['repository'].upper() == 'UNDA'.upper():
+        library_collection_code = 'UNDA ARCHV'
+    if manifest_info['repository'].upper() == 'SPEC'.upper():
+        library_collection_code = 'SPEC'
+    if manifest_info['repository'].upper() == 'Snite'.upper():
+        library_collection_code = 'SNITE'
+    if library_collection_code == "":
+        library_collection_code = 'SNITE'
+    return library_collection_code
+
+
 def _get_display_library(manifest_info):
+    """ Get human-readable library given repository """
     display_library = ""
-    if manifest_info['library'].upper() == 'UNDA'.upper():
+    if manifest_info['repository'].upper() == 'UNDA'.upper():
```

Comment:
Again this could be rewritten to use a dictionary like shown in the review above. I would at least recommend changing from all if's to if/elif/else